src/modes/ModeEstimation.jl

```diff
@@ -46,36 +46,30 @@ DynamicPPL.childcontext(context::OptimizationContext) = context.context
 DynamicPPL.setchildcontext(::OptimizationContext, child) = OptimizationContext(child)
 
 # assume
-function DynamicPPL.tilde_assume(rng::Random.AbstractRNG, ctx::OptimizationContext, spl, dist, vn, vi)
-    return DynamicPPL.tilde_assume(ctx, spl, dist, vn, vi)
-end
-
-function DynamicPPL.tilde_assume(ctx::OptimizationContext{<:LikelihoodContext}, spl, dist, vn, vi)
-    r = vi[vn]
+function DynamicPPL.tilde_assume(ctx::OptimizationContext{<:LikelihoodContext}, dist, vn, vi)
+    r = vi[vn, dist]
     return r, 0, vi
 end
 
-function DynamicPPL.tilde_assume(ctx::OptimizationContext, spl, dist, vn, vi)
-    r = vi[vn]
+function DynamicPPL.tilde_assume(ctx::OptimizationContext, dist, vn, vi)
+    r = vi[vn, dist]
     return r, Distributions.logpdf(dist, r), vi
 end
 
 # dot assume
-function DynamicPPL.dot_tilde_assume(rng::Random.AbstractRNG, ctx::OptimizationContext, sampler, right, left, vns, vi)
-    return DynamicPPL.dot_tilde_assume(ctx, sampler, right, left, vns, vi)
-end
-
-function DynamicPPL.dot_tilde_assume(ctx::OptimizationContext{<:LikelihoodContext}, sampler::SampleFromPrior, right, left, vns, vi)
+function DynamicPPL.dot_tilde_assume(ctx::OptimizationContext{<:LikelihoodContext}, right, left, vns, vi)
     # Values should be set and we're using `SampleFromPrior`, hence the `rng` argument shouldn't
     # affect anything.
-    r = DynamicPPL.get_and_set_val!(Random.GLOBAL_RNG, vi, vns, right, sampler)
+    # TODO: Stop using `get_and_set_val!`.
+    r = DynamicPPL.get_and_set_val!(Random.GLOBAL_RNG, vi, vns, right, SampleFromPrior())
```

Comment:
Unrelated, but these `Random.GLOBAL_RNG` seem surprising and a bit fishy. Maybe at least use
```suggestion
    r = DynamicPPL.get_and_set_val!(Random.default_rng(), vi, vns, right, SampleFromPrior())
```
?

Reply:
Ah, I missed the comment above.

Reply:
But that means ideally we would not use a `SamplingContext`.

Reply:
Yeah, this `get_and_set_val!` shouldn't actually be doing any sampling, but because of how it's implemented it still requires an `rng`. It basically samples if a value is not present in `varinfo` or if we're using `SampleFromUniform`.